### DISCUSSION TITLE
has_many of super Model

### DISCUSSION THREAD

**Discussion by nemo-omen** [emmett-framework/emmett#476] [General]
I have a Model, `Entry`, from which multiple models inherit a `belongs_to` relationship to another `Upload` model. All models that inherit from `Entry` have their own fields and tables. The only thing they have in common is that each belongs to an `Upload`.

I was hoping to be able to use that inheritance to query an `Upload` for any `Entry` that belongs to it. I'm not sure how to implement this (or if it's even possible/wise). The code I have so far (below) results in a `maximum recursion exceeded` error.

I'm trying to allow users to upload different CSV files and, rather than saving the file to disk, enter the file data into the appropriate table with each record referencing the upload. This is an undergraduate project and I'm trying to save resources where I can. If I can't get this to work, I'll just take the more straightforward approach of saving the raw file to disk and adding a `file_path` to the upload instead.

Any suggestions would be appreciated!

```python

class Entry(Model):
    belongs_to('upload')

class Upload(Model):
    @has_many(field='upload')
    def entries(self):
        return Entry.where(lambda m: m.upload == self.id)
    # belongs_to('user')
    file_name = Field.text(unique=True)
    upload_date = Field.datetime()

    default_values = {
        'upload_date': now
    }

    validation = {
        'file_name': {'presence': True},
        'upload_date': {'presence': True}
    }

```

**Comment by gi0baro** [comment]
@nemo-omen I'm not sure what you're asking for is feasible. From what you describe, you would like to make a union select which brings together all the columns of every table which FKs to `Upload` dynamically.. which other than very complicated also sounds quite strange to me as an implementation 🤔 

I would probably go with a different approach, so rather:
- invert the relationship, so makes `Upload` to point to other models dynamically (kinda what you see from docs https://emmett.sh/docs/2.5.x/orm/advanced#customize-has_one-and-has_many-sets with the photos example)
- don't go with relations at all, and put an upload field directly in your `Entry` abstract model, so every model which inherits from it has its own column where to store the files contents

But it is really a matter of your application design.

**Reply by nemo-omen** [reply]
That makes a lot of sense. Thanks!
